src/core/main.c

```diff
@@ -148,10 +148,16 @@ noreturn static void crash(int sig) {
         struct sigaction sa;
         pid_t pid;
 
-        if (getpid() != 1)
+        if (getpid() != 1) {
                 /* Pass this on immediately, if this is not PID 1 */
                 (void) raise(sig);
-        else if (!arg_dump_core)
+                _exit(EXIT_FAILURE);
+        }
+
+        if (detect_container() <= 0)
+                (void) write_string_file("/proc/sys/kernel/core_pattern", "core", 0);
```

Comment:
Actually, if you could give the high-level overview of what the commit is supposed to change in the commit message.